jgrapht-core/src/main/java/org/jgrapht/alg/matching/GreedyWeightedMatching.java

```diff
@@ -92,18 +92,22 @@ public GreedyWeightedMatching(Graph<V, E> graph, double epsilon)
      * @return a matching
      */
     @Override
-    public Matching<E> getMatching()
+    public Matching<V, E> getMatching()
     {
         // sort edges in non-decreasing order of weight
         // (the lambda uses e1 and e2 in the reverse order on purpose)
         List<E> allEdges = new ArrayList<>(graph.edgeSet());
         Collections.sort(
-            allEdges,
-            (e1, e2) -> comparator.compare(graph.getEdgeWeight(e2), graph.getEdgeWeight(e1)));
+                allEdges,
```

Comment:
Good point, I'm not sure. I'll see whether I can come up with a proof. 
I did some experiments on 10000 randomly generated graphs. With the modification, about 60% of the matchings improved, about 30% of the matchings decreased in quality.